### ISSUE TITLE
Xen failing to build

### ISSUE THREAD

**Issue by Chiiruno** [NixOS/nixpkgs#47101]
## Issue description
I don't have xen explicitly installed, but I guess libvirt is pulling it or something.
```
gcc   -m32 -march=i686 -DBUILD_ID -fno-strict-aliasing -std=gnu99 -Wall -Wstrict-prototypes -Wdeclaration-after-statement -Wno-unused-but-set-variable -Wno-unused-local-typedefs   -O2 -fomit-frame-pointer -D__XEN_INTERFACE_VERSION__=__XEN_LATEST_INTERFACE_VERSION__ -MMD -MF .dsdt_anycpu_qemu_xen.o.d -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -mno-tls-direct-seg-refs  -DNDEBUG -Werror -fno-stack-protector -fno-exceptions -fno-builtin -msoft-float -I/build/xen-4.8.3/tools/firmware/hvmloader/../../../tools/include -D__XEN_INTERFACE_VERSION__=__XEN_LATEST_INTERFACE_VERSION__ -DENABLE_ROMBIOS -I../../libacpi -I. -DLIBACPI_STDUTILS=\"/build/xen-4.8.3/tools/firmware/hvmloader/util.h\"  -c -o dsdt_anycpu_qemu_xen.o dsdt_anycpu_qemu_xen.c
dsdt_anycpu.c:7495:28: error: 'dsdt_anycpu' undeclared here (not in a function); did you mean 'dsdt_anycpu_len'?
 int dsdt_anycpu_len=sizeof(dsdt_anycpu);
                            ^~~~~~~~~~~
                            dsdt_anycpu_len
make[7]: *** [/build/xen-4.8.3/tools/firmware/hvmloader/../../../tools/Rules.mk:193: dsdt_anycpu.o] Error 1
make[7]: *** Waiting for unfinished jobs....
echo "#ifdef ROM_INCLUDE_VGABIOS" >> roms.inc.new
sh ./mkhex vgabios_stdvga ../vgabios/VGABIOS-lgpl-latest.bin >> roms.inc.new
dsdt_15cpu.c:4453:27: error: 'dsdt_15cpu' undeclared here (not in a function); did you mean 'dsdt_15cpu_len'?
 int dsdt_15cpu_len=sizeof(dsdt_15cpu);
                           ^~~~~~~~~~
                           dsdt_15cpu_len
make[7]: *** [/build/xen-4.8.3/tools/firmware/hvmloader/../../../tools/Rules.mk:193: dsdt_15cpu.o] Error 1
echo "#endif" >> roms.inc.new
echo "#ifdef ROM_INCLUDE_VGABIOS" >> roms.inc.new
sh ./mkhex vgabios_cirrusvga ../vgabios/VGABIOS-lgpl-latest.cirrus.bin >> roms.inc.new
echo "#endif" >> roms.inc.new
echo "#ifdef ROM_INCLUDE_ETHERBOOT" >> roms.inc.new
sh ./mkhex etherboot ../etherboot/ipxe/src/bin/rtl8139.rom ../etherboot/ipxe/src/bin/8086100e.rom >> roms.inc.new
dsdt_anycpu_qemu_xen.c:4619:37: error: 'dsdt_anycpu_qemu_xen' undeclared here (not in a function); did you mean 'dsdt_anycpu_qemu_xen_len'?
 int dsdt_anycpu_qemu_xen_len=sizeof(dsdt_anycpu_qemu_xen);
                                     ^~~~~~~~~~~~~~~~~~~~
                                     dsdt_anycpu_qemu_xen_len
make[7]: *** [/build/xen-4.8.3/tools/firmware/hvmloader/../../../tools/Rules.mk:193: dsdt_anycpu_qemu_xen.o] Error 1
echo "#endif" >> roms.inc.new
mv roms.inc.new roms.inc
make[7]: Leaving directory '/build/xen-4.8.3/tools/firmware/hvmloader'
make[6]: *** [Makefile:68: all] Error 2
make[6]: Leaving directory '/build/xen-4.8.3/tools/firmware/hvmloader'
make[5]: *** [/build/xen-4.8.3/tools/firmware/../../tools/Rules.mk:217: subdir-all-hvmloader] Error 2
make[5]: Leaving directory '/build/xen-4.8.3/tools/firmware'
make[4]: *** [/build/xen-4.8.3/tools/firmware/../../tools/Rules.mk:212: subdirs-all] Error 2
make[4]: Leaving directory '/build/xen-4.8.3/tools/firmware'
make[3]: *** [Makefile:30: all] Error 2
make[3]: Leaving directory '/build/xen-4.8.3/tools/firmware'
make[2]: *** [/build/xen-4.8.3/tools/../tools/Rules.mk:217: subdir-install-firmware] Error 2
make[2]: Leaving directory '/build/xen-4.8.3/tools'
make[1]: *** [/build/xen-4.8.3/tools/../tools/Rules.mk:212: subdirs-install] Error 2
make[1]: Leaving directory '/build/xen-4.8.3/tools'
make: *** [Makefile:101: install-tools] Error 2
builder for '/nix/store/canf4ajxd1j7izmlvvgizc6lyvlbna3q-xen-4.8.3.drv' failed with exit code 2
cannot build derivation '/nix/store/x9r8h89kby6k8xpsdig51qjaf0ghn6al-libvirt-glib-1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/5yfan9adv55k0vmhxa2z8xcjpxnzs20x-virt-manager-1.5.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/i0cxzwlmypmdz3apy5d154qsmyx6153l-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/92xbbrj1xhs2xjq69d7ssik842zfmy2x-nixos-system-okinan-19.03pre152924.5664e64a015.drv': 1 dependencies couldn't be built
error: build of '/nix/store/92xbbrj1xhs2xjq69d7ssik842zfmy2x-nixos-system-okinan-19.03pre152924.5664e64a015.drv' failed
```

### Steps to reproduce
`virtualisation.libvirtd.enable = true`?

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.18.7, NixOS, 19.03pre152634.218ce4de508 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.1`
 - channels(root): `"nixos-19.03pre152924.5664e64a015, nixpkgs-19.03pre153060.e40d850fd13"`
 - channels(okina): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by Chiiruno** [comment]
Looks like it updated, and is building correctly now, so closing.
Thanks for the expedient response!

**Comment by Zildj1an** [comment]
> Yes, `xen` was temporarily broken but fixed again in [08b94e3](https://github.com/NixOS/nixpkgs/commit/08b94e3c4580a843b82d662d94afcff6af441f81), which hasn't reached the nixos-unstable channel yet. Should be fixed the next time the channel advances.

Could you point me to the patch for this error?
